test/hotspot/gtest/runtime/test_trim_native.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2023 Red Hat Inc. All rights reserved.
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 1997, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "runtime/os.hpp"
+#include "runtime/trimNativeHeap.hpp"
+#include "unittest.hpp"
+
+TEST_VM(TrimNative, SuspendMark) {
+
+  if (!NativeHeapTrimmer::enabled()) {
+    return;
+  }
+
+  // Try recursive pausing
+  {
+    NativeHeapTrimmer::SuspendMark sm1("Test1");
+    {
+      NativeHeapTrimmer::SuspendMark sm2("Test2");
+      {
+        NativeHeapTrimmer::SuspendMark sm3("Test3");
```

Comment:
What is this actually testing? Everything could be a no-op.

Reply:
Whether it blows up or not with internal asserts that test suspend count overflow.

But you are right; this raises another issue. One is that this is never tested since we need a gtest variant running with native trimming enabled. Another is that we need a printout of the trim state, at least for VMError::report, and that could be used here too to check the expected suspend count.